src/redux/thunks/file.ts

```diff
@@ -1154,6 +1155,64 @@ export function batchGetDirectLinks(index: number, files: FileResponse[]): AppTh
   };
 }
 
+export function resetThumbnails(files: FileResponse[]): AppThunk {
+  return async (dispatch, getState) => {
+    const cache = getCachedThumbExts();
+    const uris = files
+      .filter((f) => f.type == FileType.file)
+      .filter((f) =>
+        cache === undefined || cache === null ? true : cache.has((fileExtension(f.name) || "").toLowerCase()),
+      )
+      .map((f) => getFileLinkedUri(f));
+
+    if (uris.length === 0) {
+      enqueueSnackbar({
+        message: i18next.t("application:fileManager.noFileCanResetThumbnail"),
+        preventDuplicate: true,
+        variant: "warning",
+        action: DefaultCloseAction,
+      });
+      return;
+    }
+
+    try {
+      // Re-enable thumbnails by removing the disable mark, and update local metadata/cache.
+      const targetFiles = files
+        .filter((f) => f.type == FileType.file)
+        .filter((f) =>
+          cache === undefined || cache === null ? true : cache.has((fileExtension(f.name) || "").toLowerCase()),
+        );
+
+      await dispatch(
+        patchFileMetadata(FileManagerIndex.main, targetFiles, [
+          {
+            key: Metadata.thumbDisabled,
+            remove: true,
+          },
+        ]),
+      );
+
+      // 预取：立即为所选文件请求缩略图（不依赖列表刷新或滚动触发）
+      const fm = getState().fileManager[FileManagerIndex.main];
+      const toPrefetch = targetFiles
+        .map((f) => fm.list?.files.find((ff) => ff.path === f.path) || f)
+        .filter((f): f is FileResponse => !!f);
+      // 并发触发 GET /file/thumb
+      await Promise.allSettled(toPrefetch.map((f) => dispatch(loadFileThumb(FileManagerIndex.main, f))));
```

Comment:
`/file/thumb` is expensive. Consider not loading it proactively. Just reset the thumbnail status, and replying on user scroll in view to trigger the loading.